### PR TITLE
SmallCards & CompanyPageCards - toggle for External Pages

### LINKED ISSUE #1344: 🐛Bug - Home Page - Rules Link prefetch returns 404
### Background

This was identified, while investigating the sources of big number of failed requests in Application Insights.

The error is thrown by the prefetch request of https://www.ssw.com.au/rules/.

Next.js Link component performs a prefetch request on hover of any links that are relative or absolute with the same domain.

Because [Rules website](https://www.ssw.com.au/rules/) and [SSW website](https://www.ssw.com.au/) have the same domain, Next.js Link tries to prefetch Rules.

However, because Rules is actually its own repository, Next.js does not generate a next/data/..../rules.json file. 

Because that file does not exist, but the prefetch still happens, 404 error is logged.

### Steps to replicate:

1. Go to https://www.ssw.com.au/
2. Hover over the Rules link card
3. Open Dev Tools
4. Notice Error logged in the console
5. Open network tab
6. Notice request to rules.json fail

Image
Figure: Seems like the Link for Rules is the source of error.

### Solution

Let's convince Next.js Link that it does not need to prefetch Rules link

### PR DESCRIPTION
**Pull request by vladislav-kir**
An update to issue #1344

Next JS link prefetches any links with the same domain as the current page.
/rules and /people paths have the same domain, but are not the next js pages that can be successfully prefetched.

Adding a boolean isExternal  parameter to SmallCards and CompanyPageCards.

If a card has isExternal selected, a regular anchor tag is used, which does not perform a prefetch.

- Affected routes: 
    - https://www.ssw.com.au/
    - https://www.ssw.com.au/company
- Fixed #1344

image
Figure: External Page toggle

image
Figure: Company page - Should not prefetch /people page

image
Figure: Home page - Should not prefetch /rules page

### CODE REVIEW

File: .tina/collections/company.tsx [hunk 1]
```diff
@@ -208,6 +209,13 @@ export const companyIndexSchema: Collection = {
           label: "Page URL",
           name: companyIndexSchemaConstants.companyPages.pageURL,
         },
+        {
+          type: "boolean",
+          label: "External Page",
+          description:
+            "Please select this, if the page belongs to another repository. For example: /rules, /people",
```
Comment (bradystroud) [inline .tina/collections/company.tsx]:
```suggestion
          description:
            "Select this if the link is not part of the website. This includes SSW.Rules, and SSW.People links",
```

File: components/blocks/serviceCards.tsx [hunk 2]
```diff
@@ -315,6 +334,13 @@ export const serviceCardsBlockSchema: Template = {
           // @ts-ignore
           uploadDir: () => "service-cards",
         },
+        {
+          type: "boolean",
+          label: "External Page",
+          description:
+            "Please select this, if the page belongs to another repository. For example: /rules, /people",
```
Comment (bradystroud) [inline components/blocks/serviceCards.tsx]:
```suggestion
          description:
            "Select this if the link is not part of the website. This includes SSW.Rules, and SSW.People links",
```

**Review by bradystroud — CHANGES_REQUESTED** [review]
Looking good! Some quick changes

File: components/blocks/serviceCards.tsx (hunk 2 above)
Comment (vladislav-kir) [reply components/blocks/serviceCards.tsx]: Done

File: .tina/collections/company.tsx (hunk 1 above)
Comment (vladislav-kir) [reply .tina/collections/company.tsx]: Done

**Review by bradystroud — APPROVED** [review]
LGTM thanks @vladislav-kir

**Pull request merged**